vnext/include/ReactUWP/Views/KeyboardEventHandler.h

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include <optional>
+#include <set>
+#include <folly/dynamic.h>
+#include <winrt/Windows.UI.Xaml.Controls.h>
+#include <winrt/Windows.UI.Xaml.Input.h>
+
+#include "XamlView.h"
+#include <IReactInstance.h>
+
+namespace winrt {
+  using namespace Windows::UI;
+  using namespace Windows::UI::Xaml;
+  using namespace Windows::UI::Xaml::Controls;
+  using namespace Windows::UI::Xaml::Input;
+  using namespace Windows::Foundation;
+  using namespace Windows::UI::Xaml::Media;
+  using namespace Windows::System;
+  using namespace Windows::UI::Core;
+}
+
+using namespace std;
+
+namespace react { namespace uwp {
+  enum class HandledEventPhase
+  {
+    CAPTURING = 1, // match the value with EventPhase in React. EventPhase includes None, Capturing, AtTarget, Bubbling
+    BUBBLING = 3
+  };
+
+  struct ModifiedKeyStatue
+  {
+    bool altKey{ false };
+    bool ctrlKey{ false };
+    bool metaKey{ false };
+    bool shiftKey{ false };
+    bool capLocked{ false };
+  };
+
+  struct ReactKeyboardEvent: ModifiedKeyStatue
+  {
+    int64_t target = 0;
+    string key{};
+  };
+
+  struct KeyboardEvent: ModifiedKeyStatue
+  {
+    HandledEventPhase handledEventPhase{ HandledEventPhase::BUBBLING };
+    string key{};
+  };
+
+  typedef std::function<void(winrt::IInspectable const&, winrt::KeyRoutedEventArgs const&)> KeyboardEventCallback;
+
+  class KeyboardEventBaseHandler {
+  public:
+    KeyboardEventBaseHandler(KeyboardEventCallback keyDown, KeyboardEventCallback keyUp);
```

Comment:
>KeyboardEventCallback [](start = 29, length = 21)

should these be && args and force everything to be std::move'ing these in?  std function is not the cheapest thing to copy